lib/vast/Dialect/Unsupported/UnsupportedOps.cpp

```diff
@@ -25,15 +25,18 @@ namespace vast::unsup {
     }
 
     void UnsupportedStmt::build(
-        Builder &bld, State &st, llvm::StringRef name, Type rty, const std::vector< BuilderCallBackFn > &builders
+        Builder &bld, State &st, llvm::StringRef name, Type rty,
+        const std::vector< BuilderCallBackFn > &builders
     ) {
         InsertionGuard guard(bld);
-        st.addTypes(rty);
+        // Optional, add a check if rty exist.
+        if (rty) {
```

Comment:
Does it break anything when we store empty type? It should mean nullopt as intended in such case.

Reply:
@xlauko, Yes if the type is null, it asserts inside the llvm mlir code that checks if the type is valid.